components/aws/sagemaker/tests/integration_tests/conftest.py

```diff
@@ -180,3 +182,80 @@ def experiment_id(kfp_client, tmp_path_factory, worker_id):
             data = get_experiment_id(kfp_client)
             fn.write_text(data)
     return data
+
+
+@pytest.fixture(scope="session")
+# Deploy endpoint for testing model monitoring
+def deploy_endpoint(sagemaker_client, s3_data_bucket, sagemaker_role_arn, region):
```

Comment:
can we move it to a different file?
also these resources will be created to run any test even in canaries, how can we change that?

Reply:
Though the `deploy_endpoint` is put in conftest.py, only tests that have `deploy_endpoint` as an input parameter will actually create the resource.